src/monodroid/jni/android-system.cc

```diff
@@ -779,6 +779,11 @@ AndroidSystem::setup_environment (jstring_wrapper& name, jstring_wrapper& value)
 			knownEnvVars.MonoLLVM = true;
 			return;
 		}
+
+		if (strcmp (k, "mono.enable_assembly_preload") == 0) {
```

Comment:
...and regardless of what the default *is*, this should actually *use* the value specified.  If someone does:

    mono.enable_assembly_preload=0

i.e. *explicitly disable* it via a *user-provided* `@(AndroidEnvironment)` file, then *it should be disabled*.

Reply:
Done